### PR TITLE
Feature add tag

### LINKED ISSUE #162: Update tag classes to support cumulative addition of tags

### PR DESCRIPTION
**Pull request by izzahaj**
Closes #162 
This PR modifies the add tag feature such that:
* users can now add tags cumulatively via the `addTag` command (instead of typing all tags again using `edit`)
* `edit` no longer supports editing of tags
* Tags are now stored in a `TagList` class (previously, they were stored in a `Set`)

The current command syntax is `addTag PATIENT_INDEX t/TAG`, since `AddGenericCommandParser`, which handles `-p` and other similar tags, has not been updated. Additionally, `TASK_INDEX` now uses the `-d` option (previously `-t`)

Tests for this feature have also been written.

### CODE REVIEW

**Comment by izzahaj** [comment]
I've updated the new `TagList` to always sort the tags in lexicographical order upon invocation of the `TagList` constructor.

File: src/test/java/seedu/uninurse/logic/commands/AddTagCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+        assertCommandFailure(addTagCommand, model, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_invalidDuplicateTagFilteredList_throwsCommandException() {
+        showPersonAtIndex(model, INDEX_SECOND_PERSON);
+        Patient patientToEdit = model.getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased());
+        Tag tag = new Tag("high-risk");
+        AddTagCommand addTagCommand = new AddTagCommand(INDEX_FIRST_PERSON, tag);
```
Comment (BlopApple) [inline src/test/java/seedu/uninurse/logic/commands/AddTagCommandTest.java:128]: Same as above.

File: src/test/java/seedu/uninurse/model/tag/TagListTest.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    @Test
+    public void add_duplicateTag_throwsDuplicateTagException() {
+        assertThrows(DuplicateTagException.class, () -> tagListElderly.add(TAG_ELDERLY));
+    }
+
+    @Test
+    public void add_validTag_success() {
+        TagList updatedTagList = emptyTagList.add(TAG_ELDERLY);
+        assertEquals(updatedTagList, tagListElderly);
+    }
```
Comment (BlopApple) [inline src/test/java/seedu/uninurse/model/tag/TagListTest.java:52]: Maybe we can add an additional test case for `TagList` specifically to ensure newly added tags will still yield a sorted `TagList`?

**Review by BlopApple — CHANGES_REQUESTED** [review]
Just one minor change in the test cases. As for the suggestions you don't have to change it now but we should probably create a separate issue addressing the quality of the test cases as that can be done in milestone v1.4.

File: src/test/java/seedu/uninurse/logic/commands/AddTagCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        AddTagCommand addTagCommand = new AddTagCommand(outOfBoundIndex, TAG_ELDERLY);
+
+        assertCommandFailure(addTagCommand, model, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_invalidDuplicateTagUnfilteredList_throwsCommandException() {
+        Patient patientToEdit = model.getFilteredPersonList().get(INDEX_SECOND_PERSON.getZeroBased());
+        Tag tag = new Tag("high-risk");
+        AddTagCommand addTagCommand = new AddTagCommand(INDEX_SECOND_PERSON, tag);
```
[the remark this answers is not in the document]
Comment (izzahaj) [reply src/test/java/seedu/uninurse/logic/commands/AddTagCommandTest.java:82]: created issue here #254

File: src/test/java/seedu/uninurse/model/tag/TagListTest.java (hunk 2 above)
Comment (izzahaj) [reply src/test/java/seedu/uninurse/model/tag/TagListTest.java:52]: Ok I've updated for this suggestion. As for the others, we can discuss it during the meeting

**Review by BlopApple — APPROVED** [review]
LGTM

**Pull request merged**